src/common/LoaderRow.js

```diff
@@ -1,43 +1,59 @@
 // @flow
 
 import type { LoadObject } from 'brewskey.js-api';
+import type { RowItemProps } from './SwipeableRow';
 
 import * as React from 'react';
+import { observer } from 'mobx-react';
 import ErrorListItem from './ErrorListItem';
 import LoadingListItem from './LoadingListItem';
-
-type Props<TEntity> = {|
-  renderErrorListItem: (error: Error) => React.Node,
-  renderListItem: (item: TEntity) => React.Node,
+import LoaderComponent from './LoaderComponent';
+
+// todo fix Flow in the file, it should be something else intestead *
+// for TExtraProps
+type Props<TEntity, TExtraProps = {}> = {
+  ...TExtraProps,
+  errorRow: React.ComponentType<{ ...TExtraProps, error: Error }>,
+  index: number,
+  loadedRow: React.ComponentType<RowItemProps<TEntity, TExtraProps>>,
   loader: LoadObject<TEntity>,
-  renderLoadingListItem: () => React.Node,
-|};
-
-class LoaderRow<TEntity> extends React.PureComponent<Props<TEntity>> {
-  static defaultProps = {
-    renderErrorListItem: () => <ErrorListItem />,
-    renderLoadingListItem: () => <LoadingListItem />,
-  };
-
-  render() {
-    const {
-      loader,
-      renderErrorListItem,
-      renderListItem,
-      renderLoadingListItem,
-    } = this.props;
-
-    // todo add different rows visualizations for updating/deleting states
-    if (loader.isLoading() || loader.isUpdating() || loader.isDeleting()) {
-      return renderLoadingListItem();
-    }
-
-    if (loader.hasError()) {
-      return renderErrorListItem(loader.getErrorEnforcing());
-    }
-
-    return renderListItem(loader.getValueEnforcing());
-  }
-}
+  loadingRow: React.ComponentType<TExtraProps>,
+  separators: Object,
+} & RowItemProps<TEntity, TExtraProps>;
+
+const LoaderRow = observer(
+  <TEntity>({
+    errorRow = ErrorListItem,
+    index,
+    loader,
+    loadingRow = LoadingListItem,
+    separators,
+    ...extraProps
+  }): Props<TEntity, *> => (
```

Comment:
I coldn't make Flow happy with extraProps, so use * for now, for not throttling process :(